ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -435,11 +439,11 @@ encodeShelleyResult query = case query of
     GetStakeDistribution                       -> toCBOR
     GetFilteredUTxO {}                         -> toCBOR
     GetUTxO                                    -> toCBOR
-    DebugEpochState                            -> toCBOR
+    DebugEpochState                            -> wrapCBORinCBOR toCBOR
     GetCBOR {}                                 -> encode
     GetFilteredDelegationsAndRewardAccounts {} -> toCBOR
     GetGenesisConfig                           -> toCBOR
-    DebugNewEpochState                         -> toCBOR
+    DebugNewEpochState                         -> wrapCBORinCBOR toCBOR
```

Comment:
Duncan on Slack explained that the cost of introducing CBOR-in-CBOR for `Debug` is that it'll break all current clients that use the the `GetCBOR` approach. So we're all trying to figure out how to avoid that now.

The motivator for this change is that ledger does not want to expose a `Decoder s EpochState`, it only wants to expose `Decoder s (Annotator EpochState)`, which requires a CBOR-in-CBOR.

In particular, `Decoder` is not expressive enough to support a combinator that converts `Decoder s (Annotator EpochState)` (which assumes CBOR-in-CBOR) to a `Decoder s EpochState` (which does not).

This case expression is ultimately used to instantiate the Consensus Layer's `decodeResult` class, which requires a `Decoder`. If we generalized `decodeResult`, we could perhaps enable such a combinator.